src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -271,26 +283,115 @@ private BoundInterpolatedString BindUnconvertedInterpolatedStringToHandlerType(B
             conversionDiagnostics?.Free();
 
             var intType = GetSpecialType(SpecialType.System_Int32, diagnostics, unconvertedInterpolatedString.Syntax);
-            var arguments = ImmutableArray.Create<BoundExpression>(
-                new BoundLiteral(unconvertedInterpolatedString.Syntax, ConstantValue.Create(baseStringLength), intType) { WasCompilerGenerated = true },
-                new BoundLiteral(unconvertedInterpolatedString.Syntax, ConstantValue.Create(numFormatHoles), intType) { WasCompilerGenerated = true });
+            int constructorArgumentLength = 2 + additionalConstructorArguments.Length;
+            var argumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(constructorArgumentLength);
+            populateArguments(unconvertedInterpolatedString.Syntax, additionalConstructorArguments, baseStringLength, numFormatHoles, intType, argumentsBuilder);
+
+            var refKindsBuilder = ArrayBuilder<RefKind>.GetInstance(constructorArgumentLength);
+            refKindsBuilder.Add(RefKind.None);
+            refKindsBuilder.Add(RefKind.None);
+            refKindsBuilder.AddRange(additionalConstructorRefKinds);
+
+            var nonOutConstructorDiagnostics = BindingDiagnosticBag.GetInstance(withDiagnostics: true, withDependencies: diagnostics.AccumulatesDependencies);
+            BoundExpression constructorCall;
+            BoundExpression nonOutConstructorCall = MakeConstructorInvocation(interpolatedStringHandlerType, argumentsBuilder, refKindsBuilder, unconvertedInterpolatedString.Syntax, nonOutConstructorDiagnostics);
```

Comment:
If I'm reading the spec right, I'd expect we try to find  the constructor with final `out bool` parameter first, then retry without it.
[https://github.com/dotnet/csharplang/blob/main/proposals/improved-interpolated-strings.md#constructor-resolution](https://github.com/dotnet/csharplang/blob/main/proposals/improved-interpolated-strings.md#constructor-resolution)

Reply:
I need to reverse the order, thanks for pointing it out.

Reply:
Actually, thinking about it more I had a reason for doing that in the order I put it in the spec. I'll reverse this.